### PR TITLE
Implement to_list() for DAOs (GSI-2465)

### PR DESCRIPTION
**Pull request by TheByronHimes**
This PR implements `.to_list()` for the `FindResult` class.

No version bump included because unreleased changes that are already merged to main have the requisite version bump.

### CODE REVIEW

File: tests/integration/providers/dao/test_mongodb_dao.py [hunk 1]
```diff
@@ -972,3 +972,20 @@ async def test_dao_find_all_total_count(mongodb: MongoDbFixture):
...
+        id_field="id",
+    )
+
+    resources = [ExampleDto(field_b=i) for i in range(3)]
+    for resource in resources:
+        await dao.insert(resource)
+
+    result = dao.find_all(mapping={}, sort=["field_b"])
+    items = await result.to_list()
+    assert [item.field_b for item in items] == [0, 1, 2]
```
Comment (Cito) [inline tests/integration/providers/dao/test_mongodb_dao.py]:
```suggestion
    assert isinstance(items, list)
    assert items == resources
```

to make the test stricter and show more clearly what we expect

File: tests/integration/providers/daopubsub/test_mongokafka_pubsub.py [hunk 2]
```diff
@@ -1040,3 +1040,25 @@ async def test_pagination_against_flaky_metadata(
...
+            event_topic=EXAMPLE_TOPIC,
+        )
+
+        dtos = [ExampleDto(field_b=i) for i in range(3)]
+        for dto in dtos:
+            await dao.insert(dto)
+
+        result = dao.find_all(mapping={}, sort=["field_b"])
+        items = await result.to_list()
+        assert [item.field_b for item in items] == [0, 1, 2]
```
Comment (Cito) [inline tests/integration/providers/daopubsub/test_mongokafka_pubsub.py]:
```suggestion
        assert isinstance(items, list)
        assert items == dtos
```

File: tests/unit/test_in_mem_dao.py [hunk 3]
```diff
@@ -673,6 +673,18 @@ async def test_find_all_total_count():
...
+async def test_find_all_to_list():
+    """Test that to_list() collects all results into a list."""
+    dao = DaoClass()
+    titles = ["Apple", "Banana", "Cherry"]
+    for title in titles:
+        await dao.insert(InventoryItem(title=title, count=1))
+
+    result = dao.find_all(mapping={}, sort=["title"])
+    items = await result.to_list()
+    assert [item.title for item in items] == ["Apple", "Banana", "Cherry"]
```
Comment (Cito) [inline tests/unit/test_in_mem_dao.py]:
```suggestion
    inserted = [InventoryItem(title=title, count=1) for title in titles]
    for item in inserted:
        await dao.insert(item)

    result = dao.find_all(mapping={}, sort=["title"])
    items = await result.to_list()
    assert isinstance(items, list)
    assert items == inserted
```

File: src/hexkit/protocols/dao.py [hunk 4]
```diff
@@ -170,6 +170,10 @@ async def total_count(self) -> int:
             self._cached_total = await self._get_total_count()
         return self._cached_total
 
+    async def to_list(self) -> list[Dto]:
+        """Collect all results into a list."""
```
Comment (Cito) [inline src/hexkit/protocols/dao.py]: In the docstring we should warn that this consumes the async iterator. So if you call it twice, you will get an empty list.

File: tests/integration/providers/dao/test_mongodb_dao.py (hunk 1 above)
Comment (Cito) [reply tests/integration/providers/dao/test_mongodb_dao.py]: Maybe at this point also test that `total_count()`  shows the right number.

More edge cases that could be tested: empty list and usage together with pagination.

**Review by Cito — CHANGES_REQUESTED** [review]
Some suggestions below.

File: src/hexkit/protocols/dao.py (hunk 4 above)
Comment (TheByronHimes) [reply src/hexkit/protocols/dao.py]: Done in [27a9ce5](https://github.com/ghga-de/hexkit/pull/234/commits/27a9ce5e8f1bea598aa434e843658a0cadaaeee0)

File: tests/integration/providers/dao/test_mongodb_dao.py (hunk 1 above)
Comment (TheByronHimes) [reply tests/integration/providers/dao/test_mongodb_dao.py]: Added

**Review by Cito — APPROVED** [review]
